### PR TITLE
VOMS-Dirac: handling the nickname attribute

### PR DESCRIPTION
**Pull request by sustr4**
This should be ready for production, except the path to local cache directory is hard-coded but there is no mechanisms that creates it.

Where is that addressed?

### CODE REVIEW

File: slave/process-voms-dirac/bin/process-voms_dirac.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+PROTOCOL_VERSION='3.1.0'
+
+function process {
+
+	EXECSCRIPT="${LIB_DIR}/${SERVICE}/process-voms_dirac.pl"
+
+	create_lock
+
+	FROM_PERUN="${WORK_DIR}/voms"
```
Comment (stavamichal) [inline slave/process-voms-dirac/bin/process-voms_dirac.sh]: This directory will be also voms_dirac.

**Comment by sustr4** [comment]
https://github.com/CESNET/perun-services/pull/179/commits/afb356137e151d690ad44af881d56e78ca10d0f7#diff-1ea59bf00043abb391998e276e322c8eR16
https://github.com/CESNET/perun-services/pull/179/commits/afb356137e151d690ad44af881d56e78ca10d0f7#diff-1ea59bf00043abb391998e276e322c8eR331
https://github.com/CESNET/perun-services/pull/179/commits/afb356137e151d690ad44af881d56e78ca10d0f7#diff-1ea59bf00043abb391998e276e322c8eR174

File: slave/process-voms-dirac/lib/process-voms_dirac.pl [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+	my @attributeClassesToCreate = array_minus( @attributeClasses_toBe, @attributeClasses_current );
+
+	my @attributesToDelete = array_minus_deep( @attributes_current, @attributes_toBe );
+	my @attributesToSet = array_minus_deep( @attributes_toBe, @attributes_current );
+
+	my %membersToAdd;
+	my %membersToAdd;
+	my %rolesToAssign;
+	my %rolesToDismiss;
+        foreach $group (@groups_toBe) {
```
Comment (stavamichal) [inline slave/process-voms-dirac/lib/process-voms_dirac.pl]: Indentation.

File: slave/process-voms-dirac/lib/process-voms_dirac.pl [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+		@{$membersToRemove{"$group"}} = array_minus_deep(@{$membersToRemove{"$group"}}, @{$membersToRemove{"/$name"}}) unless( "$group" eq "/$name" ); # No need to remove user from groups if they are going to be fully removed
+		@{$membersToAdd{"$group"}} = array_minus_deep(@{$groupMembers_toBe{"$group"}}, @{$groupMembers_current{"$group"}});
+		foreach $role (@roles_toBe) {
+			@{$rolesToAssign{"$group"}{"$role"}} = array_minus_deep(@{$groupRoles_toBe{"$group"}{"$role"}}, @{$groupRoles_current{"$group"}{"$role"}});
+			@{$rolesToDismiss{"$group"}{"$role"}} = array_minus_deep(@{$groupRoles_current{"$group"}{"$role"}}, @{$groupRoles_toBe{"$group"}{"$role"}});
+			@{$rolesToDismiss{"$group"}{"$role"}} = array_minus_deep(@{$rolesToDismiss{"$group"}{"$role"}}, @{$membersToRemove{"/$name"}}); # No need to revoke roles if the user is going to be fully removed
+			@{$rolesToDismiss{"$group"}{"$role"}} = array_minus_deep(@{$rolesToDismiss{"$group"}{"$role"}}, @{$membersToRemove{"$group"}}); # No need to revoke roles if the user is going to be removed from the group
+		}
+        }
```
Comment (stavamichal) [inline slave/process-voms-dirac/lib/process-voms_dirac.pl]: Indentation.

File: slave/process-voms-dirac/lib/process-voms_dirac.pl [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+	s/^\s*Role=// for @roles_current;
+
+	our @cas=`voms-admin --vo \Q${name}\E list-cas`;
+	if ( $? != 0 ) {
+		syslog LOG_ERR, "Failed listing known CAs for VO \"$name\". Error Code $?, original message from voms-admin: @groups_current";
+		print STDERR "Failed listing known CAs for VO \"$name\". Error Code $?, original message from voms-admin: @groups_current\n";
+		$retval = 1;
+		next;
+	}
+	chomp(@cas);
```
Comment (sustr4) [inline slave/process-voms-dirac/lib/process-voms_dirac.pl]: No, not expected. Of course we are wrapping around a CLI, which can change any moment. But that's another story.

File: slave/process-voms-dirac/lib/process-voms_dirac.pl [hunk 5]
```diff
@@ -0,0 +1,352 @@
...
+use JSON::XS;
+use File::Basename;
+use File::Path qw/make_path/;
+my $vos = XMLin( '-',
+	ForceArray => [ 'role', 'group', 'user', 'vo' ],
+#	GroupTags => { role => 'roles', groups => 'group', users => 'user', vos => 'vo' },
+	KeyAttr => [] );
+my $csv = Text::CSV->new({ sep_char => ',' });
+
+$attributeStatusFilePrefix = "/var/lib/perun-services/process-voms-attributes-";
```
Comment (stavamichal) [inline slave/process-voms-dirac/lib/process-voms_dirac.pl]: This path will be: /var/lib/perun/process-voms-dirac/ and there should be the cache file.

**Comment by stavamichal** [comment]
Squashed and rebased in pull request #223 
I am closing this one.

**Pull request closed without merging**